plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -145,18 +171,44 @@ export class TeamManager {
                 }
             }
         }
+    }
 
+    private async syncEventProperties(team: Team, event: string, propertyKeys: string[]) {
+        if (!this.propertyCounterTeams.has(team.id)) {
+            return
+        }
+        const key = JSON.stringify([team.id, event])
+        let properties = this.eventPropertiesCache.get(key)
+        if (!properties) {
+            properties = new Set()
+            this.eventPropertiesCache.set(key, properties)
+        }
+        for (const property of propertyKeys) {
+            if (!properties.has(property)) {
+                properties.add(property)
+                await this.db.postgresQuery(
```

Comment:
It we could do this insert in a batch instead of per-property :thinking:

Reply:
That's pretty much what I was doing in the original implementation, leading to the deadlocks mentioned above. Even if we go from `ON CONFLICT DO UPDATE` to `ON CONFLICT DO NOTHING`, I'm going to have to implement code to retry the transactions if the block deadlocks, and all other complexities that come with it.

It would much safer since we're no longer updating the rows that are already in the database, and not running all flush operations at precisely the same moment on every thread, but these deadlocks can't be avoided --> the same new event may come in 10 times in the same `millisecond`, be handled by 4 threads, which all insert the same properties at precisely the same moment.

Thus, while a bit slower, I think ~20 simple inserts for each new event seen will scale much better.